functioncall/interface/src/test/solidity/HiddenParamDestTest.sol

```diff
@@ -14,61 +14,62 @@
  */
 pragma solidity >=0.8;
 
-import "../../main/solidity/HiddenParameters.sol";
+import "../../main/solidity/NonAtomicHiddenAuthParameters.sol";
+import "../../main/solidity/AtomicHiddenAuthParameters.sol";
 
-contract HiddenParamDestTest is HiddenParameters {
+contract HiddenParamDestTest is NonAtomicHiddenAuthParameters, AtomicHiddenAuthParameters {
     uint256 private e1;
     uint256 private e2;
-    uint256 private e3;
+    address private e3;
 
     event AllGood(bool happy);
 
     constructor(
         uint256 _expected1,
         uint256 _expected2,
-        uint256 _expected3
+        address _expected3
     ) {
         e1 = _expected1;
         e2 = _expected2;
         e3 = _expected3;
     }
 
     function funcNoParams() external {
-        (uint256 a1, uint256 a2, uint256 a3) = extractThreeHiddenParams();
+        (uint256 a1, uint256 a2, address a3) = decodeAtomicAuthParams();
         check3(a1, a2, a3);
         emit AllGood(true);
     }
 
     function funcOneParam(uint256 _val) external {
-        (uint256 a1, uint256 a2, uint256 a3) = extractThreeHiddenParams();
+        (uint256 a1, uint256 a2, address a3) = decodeAtomicAuthParams();
         check3(a1, a2, a3);
         require(_val == 17, "Error: Val");
         emit AllGood(true);
     }
 
     function funcTwoParams(uint256 _val1, uint256 _val2) external {
-        (uint256 a1, uint256 a2, uint256 a3) = extractThreeHiddenParams();
+        (uint256 a1, uint256 a2, address a3) = decodeAtomicAuthParams();
         check3(a1, a2, a3);
         require(_val1 == 17, "Error: Val1");
         require(_val2 == 23, "Error: Val2");
         emit AllGood(true);
     }
 
     function twoParamFuncNoParams() external {
-        (uint256 a1, uint256 a2) = extractTwoHiddenParams();
+        (uint256 a1, address a2) = decodeNonAtomicAuthParams();
         check2(a1, a2);
         emit AllGood(true);
     }
 
     function twoParamFuncOneParam(uint256 _val) external {
-        (uint256 a1, uint256 a2) = extractTwoHiddenParams();
-        check2(a1, a2);
+        (uint256 a1, address a3) = decodeNonAtomicAuthParams();
```

Comment:
there seems to be a slight inconsistency in what we call the second variable here. We refer to it as `a3` here, which might make sense when later compared to `e3` in `check2()`, but lines 59 and 72 refer to the address as `a2`. Is there a reason for this that I am missing?